src/wallet/wallet.cpp

```diff
@@ -4021,11 +4021,17 @@ bool CWallet::ApplyMigrationData(MigrationData& data, bilingual_str& error)
     m_internal_spk_managers.clear();
 
     // Setup new descriptors
-    SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+    SetWalletFlag(WALLET_FLAG_DESCRIPTORS | WALLET_FLAG_GLOBAL_HD_KEY);
```

Comment:
Great catch! Should we also set if for watch only wallets? On line 4190?

Reply:
It doesn't make a difference for watchonly wallets, but since we are setting it for all descriptor wallets regardless of watchonly, I suppose so.

Done.